### PR TITLE
 Migrate TechDocs to new Frontend System

### PR DESCRIPTION
**Pull request by tudi2d**
This is part of the work related to #18372. After this PR TechDocs should be configurable through config using the new Frontend System. 

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [ ] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/techdocs/src/alpha.tsx
Comment (Rugvip) [inline plugins/techdocs/src/alpha.tsx:119]: TODO: discuss naming patterns 😁

**Review by Rugvip — CHANGES_REQUESTED** [review]
👍, just needs a changeset

File: plugins/techdocs/src/alpha.tsx
Comment (tudi2d) [reply plugins/techdocs/src/alpha.tsx:119]: Should we discuss this with the TechDocs Maintainers or just at stand-up? :)

**Comment by tudi2d** [comment]
Changeset added, but running into the following errors in the `app-next`. A bit suprised about the fetchApiRef, but expected the error with the catalog.

Question: Should I add it to the `app-next` even though it's not functioning yet?

Screenshot 2023-09-27 at 14 18 10
Screenshot 2023-09-27 at 14 18 01

**Comment by tudi2d** [comment]
Blocked by finishing the routing system implementation for now

File: plugins/techdocs/src/alpha.tsx
Comment (tudi2d) [reply plugins/techdocs/src/alpha.tsx:154]: So just for reference - I just took the existing naming from the TechDocs `plugin.tsx`

File: plugins/techdocs/src/alpha.tsx
Comment (camilaibs) [inline plugins/techdocs/src/alpha.tsx:95]: Suggestion (non-blocking):
Do addons fall under this pull request? It might be a follow-up pull request, but when thinking about how adopters could install `TechDocs` addons, a suggestion is to inject them as inputs to the reader's page 🙂

File: plugins/techdocs/src/alpha.tsx
Comment (camilaibs) [inline plugins/techdocs/src/alpha.tsx:103]: Suggestion (non-blocking):
Omit `api` property, for example:
https://github.com/backstage/backstage/blob/cbed529916fa87ac0f1145d5f9a1340170dfb838/plugins/catalog/src/alpha.tsx#L36-L46

File: plugins/techdocs/src/alpha.tsx
Comment (camilaibs) [reply plugins/techdocs/src/alpha.tsx:154]: I've been using camelcase for the `Search` and `Catalog` plugins, so I'm also interested in knowing the best naming pattern 🙂

File: plugins/techdocs/src/alpha.tsx
Comment (camilaibs) [inline plugins/techdocs/src/alpha.tsx:76]: Are there any properties that could be mapped to config schemas? My suggestion would be to start with the `initialFilter`, similar to what we are experimenting with in the `Catalog` plugin:
https://github.com/backstage/backstage/blob/18711f65c2f92a153273af3bcb9de4846f25abdb/plugins/catalog/src/alpha.tsx#L130-L143

**Review by camilaibs — APPROVED** [review]
Fantastic progress 👏🏻 ! I left non-blocking suggestions, feel free to address them in a follow-up pull request if they make sense to you 🙌🏻

File: packages/app-next/src/App.tsx
Comment (camilaibs) [inline packages/app-next/src/App.tsx:53]: I believe this is not necessary, installing `@backstage/plugin-techdocs`package should be enough:

```suggestion
  plugins: [graphiqlPlugin, pagesPlugin, techRadarPlugin],
```

File: plugins/techdocs/src/alpha.tsx
Comment (camilaibs) [inline plugins/techdocs/src/alpha.tsx:46]: Suggestion (non-blocking): 
Use existing route references.
https://github.com/backstage/backstage/blob/6feb4c7d66da6ad34754aa4064d09163db4451d5/plugins/techdocs/src/routes.ts#L19-L26

File: plugins/techdocs/src/alpha.tsx
Comment (tudi2d) [reply plugins/techdocs/src/alpha.tsx:76]: I'm skipping this for now as `configSchema` for `createPageExtension` requires a `path`. Not sure why it is required & doesn't go with the default, but it would require a config change so I will skip it for now.

File: plugins/techdocs/src/alpha.tsx
Comment (tudi2d) [reply plugins/techdocs/src/alpha.tsx:86]: I don't think so for now. Should I create a seperate issue?

File: plugins/techdocs/src/alpha.tsx
Comment (tudi2d) [reply plugins/techdocs/src/alpha.tsx]: Ok - updated naming now. Lmk if it looks alright.

File: plugins/techdocs/src/alpha.tsx
Comment (camilaibs) [reply plugins/techdocs/src/alpha.tsx:86]: I would recommend creating a separate issue, it will be a good exercise to test the extensions framework 🙌🏻

File: plugins/techdocs/src/alpha.tsx
Comment (camilaibs) [inline plugins/techdocs/src/alpha.tsx:83]: I believe this id should be the same as the routeRef id, otherwise you will get a route error...

Comment (Rugvip) [reply plugins/techdocs/src/alpha.tsx:83]: Yep, this. Let's figure something out this afternoon to make this not an issue though...

**Review by Rugvip — APPROVED** [review]

File: .changeset/modern-owls-mate.md
Comment (Rugvip) [inline .changeset/modern-owls-mate.md:5]:
```suggestion
Added experimental support for declarative integration via the `/alpha` subpath.
```

I like calling this out as well :p

File: plugins/techdocs/src/alpha.tsx
Comment (tudi2d) [reply plugins/techdocs/src/alpha.tsx:46]: ok - let's keep it until we solve the id issue :)

**Review by freben — APPROVED** [review]
Nice! Are you looking into camila's comment about configSchema too?

**Review by alexlorenzi — APPROVED** [review]
👍

**Comment by tudi2d** [comment]
> Nice! Are you looking into camila's comment about configSchema too?

Did not include it yet - let's change it when adjusting the routeRef :)

**Pull request merged**
